Update Metabase banner upsell UI

WiNloSt opened the pull request:
Closes https://github.com/metabase/metabase/issues/45407

### Description

Update Metabase banner upsell UI to match the design

### How to verify

Run Metabase in OSS edition

1. Sign in as an admin
1. To go a dashboard or a question.
1. Sharing > Embed > Static embed > Look and Feel
1. See that the Metabase banner Upsell UI has been updated to match the [design](https://www.figma.com/design/1YcsFf5yYdFZTguwoKVdkj/Re-design-the-transparent-background-setting-in-interactive-embedds?node-id=917-6518&t=LcLo2AyM8ZhbZt1S-0)

### Demo
![image](https://github.com/metabase/metabase/assets/1937582/f889746e-4239-4e3f-8741-dbb4c4e5f68d)


### Checklist

- [x] Tests have been added/updated to cover changes in this PR

WiNloSt committed:
Prepare folder for nested components

WiNloSt committed:
Update upsell UI

- add `fullWidth` variant
- align title to the left

WiNloSt committed:
Fix Upsell banner having horizontal scroll on full width

WiNloSt committed:
Add new Metabase banner upsell component

WiNloSt committed:
Fix unit tests

WiNloSt commented on frontend/src/metabase/admin/upsells/UpsellHosting.tsx:
```diff
@@ -21,6 +21,7 @@ export const UpsellHosting = ({ source }: { source: string }) => {
       buttonLink="https://www.metabase.com/cloud"
       illustrationSrc={RocketGlobeIllustrationSrc}
       source={source}
+      maxWidth={202}
```
This behaves the same as previously we set `max-height: 200px` and `box-sizing: content-box`

I need this change because `box-sizing: content-box` causes horizontal scrolling to appear when used in the embed modal and changing this to `box-sizing: border-box` wraps the title.
![Screenshot 2024-07-11 at 8 10 09 PM](https://github.com/metabase/metabase/assets/1937582/80e3479d-6db9-4b6b-adfc-52542dbaf795)

iethree replied:
makes sense to me, I'm not sure we need to care all that much about text wrapping, since it will look different in all locales.

oisincoveney replied:
Yeah, I agree that we shouldn't worry too much, especially if it means that we have to use magic numbers.

On that note, could you use a constant (i.e. `UPSELL_BADGE_WIDTH = 202` or something) and make a comment about it? Otherwise someone else will remove it

WiNloSt commented on frontend/src/metabase/admin/upsells/UpsellHosting.tsx:
```diff
@@ -44,6 +45,7 @@ export const UpsellHostingUpdates = ({ source }: { source: string }) => {
       buttonLink="https://www.metabase.com/cloud"
       illustrationSrc={RocketGlobeIllustrationSrc}
       source={source}
+      maxWidth={202}
```
As explained in the above comment, it used to wrap after I changed `box-sizing`
![Screenshot 2024-07-11 at 8 10 11 PM](https://github.com/metabase/metabase/assets/1937582/7e1ae3fa-e4ca-41e5-b73f-24a6af7e7bba)

WiNloSt commented on frontend/src/metabase/admin/upsells/components/UpsellCard.tsx:
```diff
@@ -18,6 +18,20 @@ type UpsellCardProps = {
+  fullWidth?: never;
+}
+
+type Variants = FullWidthVariant | FixWidthVariant;
```
This is a very nice way to ensure we don't mistakenly pass both `maxWidth` and `fullWidth` at the same time. It's like an enum, but we also need value from `maxWidth`.

Another potential API could be
```ts
interface Variant {
  width?: number | "fullwidth"
}
```

Let me know what you think.

iethree replied:
I like what you've done with `never` here ❤️

WiNloSt commented on frontend/src/metabase/admin/upsells/components/UpsellCard.tsx:
```diff
@@ -38,9 +54,13 @@ export const _UpsellCard = ({
+    >
       {illustrationSrc && <Image src={illustrationSrc} w="100%" />}
-      <Flex gap="sm" justify="center" p="1rem" pb="0.75rem">
+      <Flex gap="sm" p="1rem" pb="0.75rem">
```
The [upsell design](https://www.figma.com/design/kdZZogH2tWBVwnG03N7XXc/Upsell-System?t=WNRRu8GYlBRcGbFm-0) all have the title left-aligned. I think this was mistakenly implemented.

WiNloSt commented on frontend/src/metabase/admin/upsells/components/Upsells.styled.tsx:
```diff
@@ -55,9 +56,24 @@ export const UpsellCTALink = styled(ExternalLink)`
+      : css`
+          max-width: 200px;
+        `}
+  box-sizing: border-box;
```
`box-sizing: content-box` with `width: 100%` causes horizontal scroll in embedding modal.

Here's when I use `box-sizing: border-box`
![image](https://github.com/metabase/metabase/assets/1937582/cc02e294-3ebf-4c22-96b5-dcaaf1251276)

WiNloSt replied:
Fixed in https://github.com/metabase/metabase/pull/45424/commits/5d078c5d19dfbac003b5c6c612ab126169bd19f4

Thank you! I didn't think much, but it seems like we have some default CSS set already and `box-sizing: border-box` is one of them, so this is fine to be removed.

WiNloSt commented on frontend/src/metabase/admin/upsells/components/Upsells.styled.tsx:
```diff
@@ -55,9 +56,24 @@ export const UpsellCTALink = styled(ExternalLink)`
+      ? css`
+          width: 100%;
+        `
+      : maxWidth
```
I think it makes sense to accept `maxWidth` if we don't want some title to wrap.

![Screenshot 2024-07-11 at 8 10 09 PM](https://github.com/metabase/metabase/assets/1937582/2b38f72d-f4c1-4fba-b4f6-9a2406a80164)
![Screenshot 2024-07-11 at 8 10 11 PM](https://github.com/metabase/metabase/assets/1937582/1b83df2a-222c-4488-b01f-cfd41782a80a)

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings/LookAndFeelSettings.tsx:
```diff
@@ -25,19 +25,20 @@ import {
 type ThemeOptions = typeof THEME_OPTIONS[number]["value"];
 
-export interface AppearanceSettingsProps {
+interface AppearanceSettingsProps {
```
We never use this type outside this file.

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings/LookAndFeelSettings.tsx:
```diff
@@ -208,18 +209,9 @@ export const LookAndFeelSettings = ({
-              >{t`a paid plan`}</ExternalLink>
-            )} to remove the banner.`}</Text>
-          </StaticEmbedSetupPaneSettingsContentSection>
+          <div aria-label={t`Removing the banner`}>
```
Adding a label since `StaticEmbedSetupPaneSettingsContentSection` also has a label for the section and it's used to target inside UI in some tests.

https://github.com/metabase/metabase/blob/7eaff412a3ad4cb509986f86bd340292647472b0/frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx#L503-L507

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings/UpsellMetabaseBanner.tsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <UpsellCard
+      title={t`Removing the banner`}
+      buttonLink="https://www.metabase.com/upgrade"
```
UTM tags are automatically added by `UpsellCard`

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:
update the copy to match the new design

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:
```diff
@@ -492,27 +492,23 @@ describe("Static Embed Setup phase", () => {
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/upgrade?utm_media=static-embed-settings-appearance&utm_source=oss",
+          "https://www.metabase.com/upgrade?utm_source=product&utm_medium=upsell&utm_campaign=remove-mb-branding&utm_content=static-embed-settings-look-and-feel&source_plan=oss",
```
These UTM tags are from the Upsell component
https://github.com/metabase/metabase/blob/d79730195303931bd4ba11f12aeb00ed77b6bf3b/frontend/src/metabase/admin/upsells/components/use-upsell-link.ts#L16-L20

WiNloSt commented:
@iethree I requested your review because I modified the Upsell UI. [More context here](https://metaboat.slack.com/archives/C063Q3F1HPF/p1720698580704139?thread_ts=1720099855.742039&cid=C063Q3F1HPF).

iethree commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings/UpsellMetabaseBanner.tsx:
I think we want to put this component in admin/upsells, and import it from there.

see: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/admin/upsells/readme.md

WiNloSt replied:
Good point. But not all upsell UIs are only shown in admin settings though. I feel like `/admin/*` should be for components only shown in admin settings. But I'll follow your suggestion since it's better to have them colocated.

iethree replied:
I think the idea of /admin/ is not just for admin settings but admin-related components

WiNloSt replied:
Fixed in https://github.com/metabase/metabase/pull/45424/commits/4232fbfcf6e015c9d49d75c036bf7447b1237b30

iethree commented on frontend/src/metabase/admin/upsells/components/UpsellCard.tsx:
```diff
@@ -18,6 +18,20 @@ type UpsellCardProps = {
+  fullWidth?: boolean;
+}
+
+interface FixWidthVariant {
```
tiny naming nit

```suggestion
interface FixedWidthVariant {
```

WiNloSt replied:
Fixed in https://github.com/metabase/metabase/pull/45424/commits/06f55e011687acfa850fcb76a0811e7f3183c349

iethree reviewed (approved):
upsells style changes look good to me!

WiNloSt committed:
Address review: Fix typo

Co-authored-by: Ryan Laurie <30528226+iethree@users.noreply.github.com>

WiNloSt committed:
Address review: Move upsel UI to `admin/upsells`

Co-authored-by: Ryan Laurie <30528226+iethree@users.noreply.github.com>

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:
```diff
@@ -34,10 +35,9 @@ const THEME_OPTIONS = [
 type ThemeOptions = typeof THEME_OPTIONS[number]["value"];
 
-export interface AppearanceSettingsProps {
+interface AppearanceSettingsProps {
```
We never use this type outside this file.

WiNloSt commented on frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:
```diff
@@ -208,18 +208,9 @@ export const LookAndFeelSettings = ({
-              >{t`a paid plan`}</ExternalLink>
-            )} to remove the banner.`}</Text>
-          </StaticEmbedSetupPaneSettingsContentSection>
+          <div aria-label={t`Removing the banner`}>
```
Adding a label since `StaticEmbedSetupPaneSettingsContentSection` also has a label for the section and it's used to target inside UI in some tests.

https://github.com/metabase/metabase/blob/7eaff412a3ad4cb509986f86bd340292647472b0/frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx#L503-L507

WiNloSt commented on frontend/src/metabase/admin/upsells/UpsellMetabaseBanner.tsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <UpsellCard
+      title={t`Removing the banner`}
+      buttonLink="https://www.metabase.com/upgrade"
```
UTM tags are automatically added by `UpsellCard`

oisincoveney commented:
I think it looks good - just two concerns from my side (not blockers, but things I'd like to bring up anyway)

*   The blue looks really off here - I don't think I've seen it around the app but I might be mistaken. That's a design thing but I was wondering if it was a deliberate choice by the designer - was there a conversation about this one?
*   I feel like we could probably do something cleaner than mess around with the `box-sizing` property and min/max widths. I'd like to play around with it (after it's merged). Maybe it something that has to do with the `display` property of the parent?

WiNloSt committed:
Remove unnecessary ESLint disable comment

We can remove it since the file is in admin folder, and is already excluded from the rule.

WiNloSt commented:
> I think it looks good - just two concerns from my side (not blockers, but things I'd like to bring up anyway)
> 
> * The blue looks really off here - I don't think I've seen it around the app but I might be mistaken. That's a design thing but I was wondering if it was a deliberate choice by the designer - was there a conversation about this one?

Here's the [design for this project](https://www.figma.com/design/1YcsFf5yYdFZTguwoKVdkj/Re-design-the-transparent-background-setting-in-interactive-embedds?node-id=917-6518&t=LcLo2AyM8ZhbZt1S-0). And the design for the [new upsell system.](https://www.figma.com/design/kdZZogH2tWBVwnG03N7XXc/Upsell-System?node-id=491-8759&t=WNRRu8GYlBRcGbFm-0)

For what it's worth, we already [have this component](https://storybook.metabase.com/?path=/story/upsells-card--without-image) and have been using it in a few places. 

I haven't been involved in the discussion of the upsell UI itself, but for this project, the direction is to [reuse the component that we already have.](https://metaboat.slack.com/archives/C063Q3F1HPF/p1720482240539049?thread_ts=1720099855.742039&cid=C063Q3F1HPF)

I kind of agree it looks a tad off here, maybe this is something we could do outside the project.

> * I feel like we could probably do something cleaner than mess around with the `box-sizing` property and min/max widths. I'd like to play around with it (after it's merged). Maybe it something that has to do with the `display` property of the parent?

To be fair, `box-sizing: border-box` is kind of standard practice since you'll have no problem with unexpected size due to border. We don't have any reference to `box-sizing: content-box` outside this component before this PR by the way.

Sounds good about width. I'm not settling on the API yet, and it could be improved, but I think limiting it to 200px shouldn't be the only option apart from the full-width version. I think we might instead want to have a size variant e.g. sm/md/lg/xl and full width instead of an arbitrary width value.

WiNloSt commented:
@oisincoveney 
> I think it looks good - just two concerns from my side (not blockers, but things I'd like to bring up anyway)
> 
> * The blue looks really off here - I don't think I've seen it around the app but I might be mistaken. That's a design thing but I was wondering if it was a deliberate choice by the designer - was there a conversation about this one?

Here's the [design for this project](https://www.figma.com/design/1YcsFf5yYdFZTguwoKVdkj/Re-design-the-transparent-background-setting-in-interactive-embedds?node-id=917-6518&t=LcLo2AyM8ZhbZt1S-0). And the design for the [new upsell system.](https://www.figma.com/design/kdZZogH2tWBVwnG03N7XXc/Upsell-System?node-id=491-8759&t=WNRRu8GYlBRcGbFm-0)

For what it's worth, we already [have this component](https://storybook.metabase.com/?path=/story/upsells-card--without-image) and have been using it in a few places. 

I haven't been involved in the discussion of the upsell UI itself, but for this project, the direction is to [reuse the component that we already have.](https://metaboat.slack.com/archives/C063Q3F1HPF/p1720482240539049?thread_ts=1720099855.742039&cid=C063Q3F1HPF)

I kind of agree it looks a tad off here, maybe this is something we could do outside the project.

> * I feel like we could probably do something cleaner than mess around with the `box-sizing` property and min/max widths. I'd like to play around with it (after it's merged). Maybe it something that has to do with the `display` property of the parent?

To be fair, `box-sizing: border-box` is kind of standard practice since you'll have no problem with unexpected size due to border. We don't have any reference to `box-sizing: content-box` outside this component before this PR by the way.

Sounds good about width. I'm not settling on the API yet, and it could be improved, but I think limiting it to 200px shouldn't be the only option apart from the full-width version. I think we might instead want to have a size variant e.g. sm/md/lg/xl and full width instead of an arbitrary width value.

WiNloSt committed:
Address review: remove default value property

Co-authored-by: Denis Berezin <denis.berezin@metabase.com>

WiNloSt commented on frontend/src/metabase/admin/upsells/components/Upsells.styled.tsx:
```diff
@@ -55,9 +56,23 @@ export const UpsellCTALink = styled(ExternalLink)`
 
-export const UpsellCardComponent = styled.div`
-  max-width: 200px;
-  box-sizing: content-box;
```
[For reference](https://github.com/metabase/metabase/pull/45424#discussion_r1674093953), box-sizing is removed so it will be `box-sizing: border-box` by default

https://github.com/metabase/metabase/blob/7e6adfaa3f3290601162135f546cdcaa3c120a85/frontend/src/metabase/css/core/box_sizing.module.css#L1-L19

heypoom reviewed (approved):
Nice! Could you extract the `maxWidth={202}` constant to a variable and add an inline comment there as Oisin suggested? Other than that this should be good to go.

WiNloSt committed:
Address review: Extract magic number
